src/engine.cpp

```diff
@@ -23,7 +23,7 @@ RandomEngine::RandomEngine():length((uint8_t)8),symb(false),mod(symb?73:63){
     //this part is independent of user's input, hence defined in the default contructor.
     std::ifstream file("/dev/urandom", std::ios::in | std::ios::binary);
     if(!file){
-    std::cout << "Error opening file\n";
+    throw std::runtime_error("Error opening file\n");
```

Comment:
What about `std::filesystem_error`? 

https://en.cppreference.com/w/cpp/filesystem/filesystem_error

Reply:
So i tried doing this:
```
    try{
    std::ifstream file("", std::ios::in | std::ios::binary);
    file.read(seq, sizeof(char)*100);
    file.close();
    }
    catch(std::filesystem::filesystem_error& ex){
        std::cerr<<ex.what()<<"\n";
    }
```
But this does not seem to throw an exception and generates a password instead.

Reply:
No it won't work because no exception is thrown. What I meant to say is that what about throwing `std::filesystem_error` instead of simple `std::runtime_error`  since it is more readable and it also inherits `std::runtime_error` ?

Reply:
Yes, it is more readable and would be better than using ```std::runtime_error```  .

Reply:
Is this how i should be doing it:
```
    std::error_code err{};
    if(!std::filesystem::exists("/dev/urandom",err)){
         throw std::filesystem::filesystem_error{ "Error opening file", err};
    }
```
this is the output if the path is invalid:
```
terminate called after throwing an instance of 'std::filesystem::__cxx11::filesystem_error'
  what():  filesystem error: Error opening file: Success
Aborted (core dumped)
```

Reply:
There is no need to check existence. just try

```
throw std::filesystem_error("Error opening file");
```

Reply:
```filesystem_error``` is a member of ```std::filesystem``` and it takes in an error code for argument as well.

Reply:
Then pass it.

Reply:
> Is this how i should be doing it:
> ```
>     std::error_code err{};
>     if(!std::filesystem::exists("/dev/urandom",err)){
>          throw std::filesystem::filesystem_error{ "Error opening file", err};
>     }
> ```
> this is the output if the path is invalid:
> ```
> terminate called after throwing an instance of 'std::filesystem::__cxx11::filesystem_error'
>   what():  filesystem error: Error opening file: Success
> Aborted (core dumped)
> ```

Ah okay, this is also fine.